pkg/broker/filter/filter_handler.go

```diff
@@ -374,7 +379,7 @@ func materializeSubscriptionsAPIFilter(ctx context.Context, filter eventingv1.Su
 		}
 	case len(filter.Suffix) > 0:
 		// The webhook validates that this map has only a single key:value pair.
-		for attribute, suffix := range filter.Exact {
+		for attribute, suffix := range filter.Suffix {
```

Comment:
Maybe we need more unit tests for this?

Reply:
I was also thinking why weren't there any tests that failed? and actually there were, the e2e experimental tests. where each filter dialect is tested. They just don't block the PR.

But yeah, I think we need more unit tests for various combinations of dialects. I'll be adding those as we move it to beta etc.